src/core/asyncUtils.ts

```diff
@@ -20,11 +20,11 @@ function asyncUtils(act: Act, addResolver: (callback: () => void) => void): Asyn
       return callbackResult ?? callbackResult === undefined
     }
 
-    const timeoutSignal = createTimeoutController(timeout)
+    const timeoutSignal = createTimeoutController(timeout as number | boolean, false)
 
     const waitForResult = async () => {
       while (true) {
-        const intervalSignal = createTimeoutController(interval)
+        const intervalSignal = createTimeoutController(interval as number | boolean, true)
```

Comment:
just as a suggestion for readability, I prefer boolean params to be named options as it makes it clearer what they do when reading from this side:

```ts
const intervalSignal = createTimeoutController(interval, { allowFakeTimers: true })
```

Reply:
👍